src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -332,9 +339,12 @@ public Task<PingReply> SendPingAsync(IPAddress address, int timeout, byte[] buff
             }
             catch (Exception e)
             {
-                Finish();
                 return Task.FromException<PingReply>(new PingException(SR.net_ping, e));
             }
+            finally
+            {
+                Finish();
+            }
```

Comment:
@wfurt, how are you validating this change?

This doesn't look right to me.  SendPingAsync calls CheckStart to set the bit, then calls SendPingAsyncCore, which kicks off the async operation, and immediately returns a Task, at which point this code calls Finish even though the operation is still likely in flight.